In [1]:
import pandas as pd
import matplotlib

In [2]:
data = pd.read_csv('clin_trials.csv')

In [3]:
data.head()

Unnamed: 0                             Organization Full Name  \
0           0                          Montefiore Medical Center   
1           1                 Seoul National University Hospital   
2           2                             Federico II University   
3           3  Sidney Kimmel Comprehensive Cancer Center at J...   
4           4                            Northwestern University   

  Organization Class       Responsible Party  \
0              OTHER                 SPONSOR   
1              OTHER                 Unknown   
2              OTHER  PRINCIPAL_INVESTIGATOR   
3              OTHER                 SPONSOR   
4              OTHER                 SPONSOR   

                                         Brief Title  \
0  Kinesiotape for Edema After Bilateral Total Kn...   
1  An Open-labeled Trial of Ramipril in Patients ...   
2         OCTA in Epivascular Glia After Dex Implant   
3  Preoperative Immune Checkpoint Inhibitor for P...   
4  Genistein in Treating Patients With Prostate C...   

                                          Full Title Overall Status  \
0  Effect of Kinesiotaping on Edema Management, P...      COMPLETED   
1  An Open-labeled Trial of Ramipril in Patients ...      COMPLETED   
2  Evaluation of Changes in Epivascular Glia Befo...      COMPLETED   
3  Preoperative Immune Checkpoint Inhibitor Thera...      COMPLETED   
4  Phase 2 Trial of Genistein in Men With Circula...     TERMINATED   

   Start Date       Standard Age  \
0  18-10-2021  ADULT OLDER_ADULT   
1     2004-10  ADULT OLDER_ADULT   
2  01-01-2021  ADULT OLDER_ADULT   
3  08-07-2019  ADULT OLDER_ADULT   
4  03-02-2011  ADULT OLDER_ADULT   

                                          Conditions Primary Purpose  \
0  Arthroplasty Complications, Arthroplasty, Repl...       TREATMENT   
1                         Migraine With Hypertension       TREATMENT   
2                               Diabetic Retinopathy         Unknown   
3  Head and Neck Squamous Cell Carcinoma, Head an...       TREATMENT   
4  Adenocarcinoma of the Prostate, Recurrent Pros...       TREATMENT   

                                       Interventions  \
0                   Kinesio(R)Tape for edema control   
1                                           Ramipril   
2                 Dexamethasone intravitreal implant   
3             Nivolumab 480mg and surgical resection   
4  genistein, placebo, therapeutic conventional s...   

                            Intervention Description      Study Type   Phases  \
0  Kinesio(R)Tape is an elastic, cotton tape with...  INTERVENTIONAL      NaN   
1                         ramipril 2.5mg twice a day  INTERVENTIONAL   PHASE2   
2                 Dexamethasone intravitreal implant   OBSERVATIONAL  Unknown   
3  One dose of Nivolumab 480mg given four weeks p...  INTERVENTIONAL   PHASE2   
4  Given orally, Given orally, Radical prostatect...  INTERVENTIONAL   PHASE2   

                                     Outcome Measure  \
0  Change from baseline and during 1-2-day time i...   
1                                 headache frequency   
2  Changes in epivascular glia after Dexamethason...   
3  Safety as measured by number of participants w...   
4  Number of Circulating Prostate Cells (CPCs) in...   

                            Medical Subject Headings  
0                                              Edema  
1                    Migraine Disorders Hypertension  
2                               Diabetic Retinopathy  
3  Carcinoma Carcinoma, Squamous Cell Head and Ne...  
4                                Prostatic Neoplasms

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 496615 entries, 0 to 496614
Data columns (total 17 columns):
 #   Column                    Non-Null Count   Dtype 
---  ------                    --------------   ----- 
 0   Unnamed: 0                496615 non-null  int64 
 1   Organization Full Name    496615 non-null  object
 2   Organization Class        496615 non-null  object
 3   Responsible Party         496615 non-null  object
 4   Brief Title               496615 non-null  object
 5   Full Title                496615 non-null  object
 6   Overall Status            496615 non-null  object
 7   Start Date                496615 non-null  object
 8   Standard Age              495730 non-null  object
 9   Conditions                496615 non-null  object
 10  Primary Purpose           496615 non-null  object
 11  Interventions             496615 non-null  object
 12  Intervention Description  496615 non-null  object
 13  Study Type                496615 non-null  object
 14  Phas

In [5]:
data.isnull().sum()

Unnamed: 0                       0
Organization Full Name           0
Organization Class               0
Responsible Party                0
Brief Title                      0
Full Title                       0
Overall Status                   0
Start Date                       0
Standard Age                   885
Conditions                       0
Primary Purpose                  0
Interventions                    0
Intervention Description         0
Study Type                       0
Phases                      183844
Outcome Measure                  0
Medical Subject Headings         0
dtype: int64

In [6]:
data.describe(include='all')

Unnamed: 0                             Organization Full Name  \
count   496615.000000                                             496615   
unique            NaN                                              28083   
top               NaN  National Institutes of Health Clinical Center ...   
freq              NaN                                               4855   
mean    248307.000000                                                NaN   
std     143360.546304                                                NaN   
min          0.000000                                                NaN   
25%     124153.500000                                                NaN   
50%     248307.000000                                                NaN   
75%     372460.500000                                                NaN   
max     496614.000000                                                NaN   

       Organization Class Responsible Party  \
count              496615            496615   
unique                  9                 4   
top                 OTHER           SPONSOR   
freq               345340            256767   
mean                  NaN               NaN   
std                   NaN               NaN   
min                   NaN               NaN   
25%                   NaN               NaN   
50%                   NaN               NaN   
75%                   NaN               NaN   
max                   NaN               NaN   

                                              Brief Title Full Title  \
count                                              496615     496615   
unique                                             494258     483008   
top     [Trial of device that is not approved or clear...    Unknown   
freq                                                  885       9990   
mean                                                  NaN        NaN   
std                                                   NaN        NaN   
min                                                   NaN        NaN   
25%                                                   NaN        NaN   
50%                                                   NaN        NaN   
75%                                                   NaN        NaN   
max                                                   NaN        NaN   

       Overall Status Start Date       Standard Age Conditions  \
count          496615     496615             495730     496615   
unique             14       8787                  6     195739   
top         COMPLETED    Unknown  ADULT OLDER_ADULT    Healthy   
freq           271022       5146             324172       8130   
mean              NaN        NaN                NaN        NaN   
std               NaN        NaN                NaN        NaN   
min               NaN        NaN                NaN        NaN   
25%               NaN        NaN                NaN        NaN   
50%               NaN        NaN                NaN        NaN   
75%               NaN        NaN                NaN        NaN   
max               NaN        NaN                NaN        NaN   

       Primary Purpose Interventions Intervention Description      Study Type  \
count           496615        496615                   496615          496615   
unique              11        360678                   383247               4   
top          TREATMENT       Unknown                  Unknown  INTERVENTIONAL   
freq            243902         49238                    75895          380931   
mean               NaN           NaN                      NaN             NaN   
std                NaN           NaN                      NaN             NaN   
min                NaN           NaN                      NaN             NaN   
25%                NaN           NaN                      NaN             NaN   
50%                NaN           NaN                      NaN             NaN   
75%                NaN           NaN                      NaN    

In [7]:
data.count()

Unnamed: 0                  496615
Organization Full Name      496615
Organization Class          496615
Responsible Party           496615
Brief Title                 496615
Full Title                  496615
Overall Status              496615
Start Date                  496615
Standard Age                495730
Conditions                  496615
Primary Purpose             496615
Interventions               496615
Intervention Description    496615
Study Type                  496615
Phases                      312771
Outcome Measure             496615
Medical Subject Headings    496615
dtype: int64

In [8]:
data2 = data.dropna()

In [9]:
data2.count()

Unnamed: 0                  311886
Organization Full Name      311886
Organization Class          311886
Responsible Party           311886
Brief Title                 311886
Full Title                  311886
Overall Status              311886
Start Date                  311886
Standard Age                311886
Conditions                  311886
Primary Purpose             311886
Interventions               311886
Intervention Description    311886
Study Type                  311886
Phases                      311886
Outcome Measure             311886
Medical Subject Headings    311886
dtype: int64

dropna reduce the data from 496615 to 311886 which give us 184729 to be dropped

In [10]:
data['Standard Age']


0         ADULT OLDER_ADULT
1         ADULT OLDER_ADULT
2         ADULT OLDER_ADULT
3         ADULT OLDER_ADULT
4         ADULT OLDER_ADULT
                ...        
496610    ADULT OLDER_ADULT
496611          CHILD ADULT
496612                ADULT
496613    ADULT OLDER_ADULT
496614    ADULT OLDER_ADULT
Name: Standard Age, Length: 496615, dtype: object

In [ ]:
c


array(['ADULT OLDER_ADULT', 'CHILD', 'ADULT', 'CHILD ADULT',
       'CHILD ADULT OLDER_ADULT', 'OLDER_ADULT', nan], dtype=object)

In [ ]:
drop_standard_age = data['Standard Age']
drop_standard_age.dropna()

0         ADULT OLDER_ADULT
1         ADULT OLDER_ADULT
2         ADULT OLDER_ADULT
3         ADULT OLDER_ADULT
4         ADULT OLDER_ADULT
                ...        
496610    ADULT OLDER_ADULT
496611          CHILD ADULT
496612                ADULT
496613    ADULT OLDER_ADULT
496614    ADULT OLDER_ADULT
Name: Standard Age, Length: 495730, dtype: object

In [15]:
count_standard_age = data['Standard Age'].value_counts()

print(count_standard_age)

Standard Age
ADULT OLDER_ADULT          324172
ADULT                       66772
CHILD ADULT OLDER_ADULT     46675
CHILD                       30015
CHILD ADULT                 20595
OLDER_ADULT                  7501
Name: count, dtype: int64
